bayesian/utility/v2/stack_analyses.py

```diff
@@ -0,0 +1,243 @@
+            error_message = "Error processing request. " \
+                            "Manifest is missing its value '{}' is " \
+                            "invalid / not supported".format(self.manifest)
+            current_app.logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # File path is mandatory and must be a string.
+        if self.file_path is None or not isinstance(self.file_path, str):
+            error_message = "Error processing request. " \
+                            "File path is missing / its value " \
+                            "'{}' is invalid".format(self.file_path)
+            current_app.logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # Ecosystem  is mandatory and must be a string.
+        if self.ecosystem is None or not isinstance(self.ecosystem, str):
+            error_message = "Error processing request. " \
+                            "Ecosystem is missing / its value '{}' " \
+                            "is invalid".format(self.ecosystem)
+            current_app.logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # Below ecosystem map tries to find the map entry for given ecosystem.
+        ecosystem_map = {
+            "node": "npm",
+            "python": "pypi",
+            "java": "maven"
+        }
+        # If given ecosystem is not found in above map, than uses the value passed in the request.
+        self.ecosystem = ecosystem_map.get(self.ecosystem, self.ecosystem)
+        current_app.logger.info("Final ecosystem after mapping is '{}'".format(self.ecosystem))
+
+        # Ecosystem should be a valid value
+        if not check_for_accepted_ecosystem(self.ecosystem):
+            error_message = "Error processing request. " \
+                            "'{}' ecosystem is not supported".format(self.ecosystem)
+            current_app.logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+    def prepare_request(self):
+        """Prepare post request by creating manifest file info and unique request id."""
+        # Build manifest info from manifest file and path. It read content in utf-8 encoding.
+        self._manifest_file_info = {
+            "filename": self.manifest.filename,
+            "filepath": self.file_path,
+            "content": self.manifest.read().decode('utf-8')
+        }
+        current_app.logger.info(self._manifest_file_info)
+
+        # Generate unique request id using UUID, also record timestamp in readable form
+        self._new_request_id = str(uuid.uuid4().hex)
+        self._date_str = str(datetime.datetime.now())
+
+    def get_request_data(self, external_new_request_id):
+        """Read request data for given request id from RDS."""
+        db_result = fetch_sa_request(rdb, external_new_request_id)
+        if db_result is None:
```

Comment:
This is already handled where this function is called.

Reply:
Point taken, I will correct it.

Reply:
This is fixed in commit 'Review comment to HTTPError from base classes is fixed.'